### ISSUE TITLE
Keener  - New Reports Needed

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1088]
In GitLab by @kdjstudios on Sep 7, 2018, 10:21

**Submitted by:** Gaylan Garrett <gaylan@keenercom.net>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-09-07-97266/conversation

**Server:** External

**Client/Site:** Keener

**Account:** NA

**Issue:** 

I was wondering if you could create 2 new reports as listed below and what cost if any would be associated with these reports.

 

## New Reports

 

An invoice Journal for all invoices issued for a date range which can include multiple invoice cycles.  The report should include:

- Account number
- Account name
- Invoice number
- Invoice date
- Amount invoiced by revenue type:
 - i.    Recur
 - ii.    Usage
 - iii.    Periodic
 - iv.    Charges
 - v.    Credits
 - vi.    Late Charges
 - vii.    Total

*This is the same information as the Revenue Analysis Worksheet by invoice cycle with the addition of the invoice # & date.*

Detail open AR report which is basically a listing of open invoices as of a month end or as of the date the report is run.  The report should include:

- Account number
- Account name
- Invoice number
- Invoice date
- Due Date
- Invoice amount
- Invoice balance

*This uses the same data as the AR Snapshot report but includes the invoice detail rather than aging the open invoice amounts.  Note that any unapplied payments or credits will also need to be included in the AR detail.*

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 7, 2018, 10:23

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 10, 2018, 06:10

An invoice Journal for all invoices issued for a date range which can include multiple invoice cycles.  The report should include:

- Account number

- Account name

- Invoice number

- Invoice date

- Amount invoiced by revenue type:

    - i.    Recur

    - ii.    Usage

    - iii.    Periodic

    - iv.    Charges

    - v.    Credits

    - vi.    Late Charges

    - vii.    Total

_This is the same information as the Revenue Analysis Worksheet by invoice cycle with the addition of the invoice # & date.


For the above request, clone the revenue analysis worksheet report and add the two columns she noted. Also, change the filter to allow running for a dare range, which means there will be multiple detail line items and a total by account will need to be added.  

-------------------------------------------------------------------
-------------------------------------------------------------------

Detail open AR report which is basically a listing of open invoices as of a month end or as of the date the report is run.  The report should include:

- Account number

- Account name

- Invoice number

- Invoice date

- Due Date

- Invoice amount

- Invoice balance

_This uses the same data as the AR Snapshot report but includes the invoice detail rather than aging the open invoice amounts.  Note that any unapplied payments or credits will also need to be included in the AR detail.


For the above request, she is looking for a report of invoices that were unpaid as of the date they specify when running the report.  It also needs to include credits that are not (or were not) yet applied to an invoice.  There may be more than one open invoice as of the date specified and there could be credits or payments that also need to be included in the report.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 10, 2018, 06:13

@kdjstudios Kyle, can you please check with them and see if they want any filters or sorting options on those reports? (We know we already have a date or date range that is needed.)  Thx

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 10, 2018, 13:26

I wrote:

> Thank you for reaching out. Our support team has started compiling an estimate for you on creating the reports. Currently they do only have one questions and that is what type of report filter parameters or sorting options would you like on each of the reports? We would presume that they will need a data range filter options.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 11, 2018, 02:11

Hi [Tim] ( @tim.traylor ) & [Kyle] ( @kdjstudios ),  

We have estimated the report development time: 

* **Invoice Journal report** - 
  * Setup - 0.5 Hr
  * Development - 1 Hr
  * Testing - 0.5 Hr
  * Buffer - 1 Hr

* **Open AR report** - 
  * Setup - 0.5 Hr
  * Development - 2 Hr
  * Testing - 0.5 Hr
  * Buffer - 1 Hr

Testing & again development if there'll be any issue - 2 Hr

Total estimate - `9 Hr`

**Note** - We're considering that the filters & sorting options are the date range & ascending name respectively. If there'll be any change in those then the estimation may vary.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 11, 2018, 02:11

added 30m of time spent at 2018-09-11

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 11, 2018, 02:11

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 13, 2018, 06:40

@smasih @abakshi Hi Atul & Sumeet,  I understand allowing some buffer time for issues, but this estimate is basically a min of 5 and a max of 9.  Saying it could take almost double the minimum time is a pretty big variance. Generally, a variance of no more than 10% of the base estimate is what I would expect to see.

I really want us to be getting more accurate estimates and I think that the team has spent enough time working on this app that they have the experience and understanding to do so.

Let's discuss this at the meeting.  Thx

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 13, 2018, 08:13

Gaylan wrote:

> Is there an update on ticket 2018-09-07-97266 for special report capabilities.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 13, 2018, 10:09

After discussion during the call today, we choose to give a 7 hour estimate.

I wrote:

> Thank you for reaching out for an update. The support team was able to review the requested reports and have just this morning been able to compile an estimate of roughly 7 hours billable time for these requests to be completed. Each report was about 3.5 hours billable time. I believe the billable service rate goes for $100 per hour. Please let us know if you approve of the billable work so we may instruct the support team to prioritize this work. Thanks and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 13, 2018, 12:25

@tim.traylor @pchaudhary @amishra @smasih 

Gaylan wrote:

> Yes, please go ahead and create the capabilities for us to run these reports. Can you give me a time frame of when they may be completed so we can start utilizing them ?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 20, 2018, 16:00

This has been marked as development for a week now. updating the priority due to the delay, as this is billable time and is a request directly from a client who is paying for this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 24, 2018, 11:04

@smasih @pchaudhary @amishra @tim.traylor 

Please bring this up for discussion in your morning meetings with @tim.traylor

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 24, 2018, 12:02

Gaylan wrote:

> Can you give me an update on the status of the creation of some reports per the email below. 

This is not good. It has been almost two weeks since they approved this and we have no updates...

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 24, 2018, 22:54

we have been busy with other tasks.

pushing to development queue.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 25, 2018, 07:30

Hi [Kyle] ( @kdjstudios ), 


1- We have also started the work on 1^st report named as **Invoice Journal** but while development we get that there is already an invoice number & date in the report. Please confirm if we're wrong.  


2- We have completed the development of 2^nd report named as **Detail Open AR**. Please have a look at this report and let us know if there is something else to do in this report. We assume that the open invoice means the invoices which are not paid yet.  

> Note that any unapplied payments or credits will also need to be included in the AR detail.  

* We would like to confirm that payments are always applied to the invoice balance & account balance, so there is no meaning to say unapplied payments, although we also would like to ask that what do they mean by the unapplied credits?  
  Do they mean to the Staged credit fees or the fees which they apply to the invoice on its draft stage?  
* If they mean only to the payments then the report is completed except the download PDF & CSV option.
* We did not added the date column till now becuase as mentioned on the report request by Gaylan that `Detail open AR report which is basically a listing of open invoices as of a month end or as of the date the report is run.` The report shows the details of the time when the report runs.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 25, 2018, 07:30

added 4h of time spent at 2018-09-25

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 25, 2018, 07:37

> Note: This branch will stay live till tomorrow morning 3am EST

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 08:43

@pchaudhary 

I am at a loss today, i don't see either of these reports on MERGE.

![image](/uploads/05fec1b3f1b9191be80fa5356f8ddbbd/image.png)

![image](/uploads/f1d131e4bf5dcce1c96aaa7a9e414bcf/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 15:45

@smasih @pchaudhary @amishra @tim.traylor 

**Invoice Journal:** Attached the export since a screenshot would not fit. [invoice_journal_07_01_2019_Master.csv](/uploads/7fde13525bb989618fa583367c9d0a8a/invoice_journal_07_01_2019_Master.csv)

- I thought this report was supposed to be ran via a date range not a billing cycle?
- I see multiple lines for each account, but I don't see the totals for each account?
- PDF export does not fit on the page
- GUI layout causes issue in the header of the web page. Maybe put this in a window with a scroll bar at the bottom and top of the report? (Same this happens on the Revenue Analysis Worksheet report too)
- What is the reference column? it seems to be blank for everything (Same this happens on the Revenue Analysis Worksheet report too)
- It does appear to be pulling the correct data in and the numbers seem to be accurate. 


**Detail Open AR:**  

![image](/uploads/28d29b2e908c0ee0a540af39b605096f/image.png)

- I thought this report was supposed to be able to select a date to run it for?
- Notice that it states "no longer includes pending and draft items". I thought it needed to include pending payments and draft credits?
- Layout appears to be good and values appear to be correct.


**ISSUE:**

So I was going to attempt to make a payment to Account Demo 54 and then rerun the report to insure that it updates accordingly. But when trying to access the payment history page I received the something went wrong error: https://merge.sabilling.com/accounts/5b658fd2cf85c153b800000a/payments?sid=

**Overall:**

It seems @pchaudhary is asking similar questions that I am. Please review this with @tim.traylor in the morning meeting to find answers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 15:51

I wrote:

> The support team has been able to create the reports and they are currently testing them. We are hoping to have these reports completely tested and ready for your review before the end of the week. I do want to give you a quick glimpse of the reports as they are right now. Please note these are using testing information and metrics; so they may appear a bit differently once we release them for you to review. However if you do see any major concerns or have any suggestions currently we would greatly appreciate your input. Thanks and have a wonderful day.  

![Invoice_Journal](/uploads/ffc867c7e9ad717cc22a0feb5e9a1daa/Invoice_Journal.png)

![Detail_Open_AR](/uploads/20c9a6bbb1d70e521c43036ce915d141/Detail_Open_AR.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 26, 2018, 24:48

Hi [Kyle] ( @kdjstudios ),   

Thanks for the replies.  

**Invoice Journal Report**:  

* We still don't get any confirmation about the **Invoice number & Date** on the Invoice Journal Report. 


> - I thought this report was supposed to be run via a date range, not a billing cycle?

* If this is the case, then we have to re-estimate of this report as the function we're using is for the billing cycle as the details are already on the database for that billing cycle.

> - I see multiple lines for each account, but I don't see the totals for each account?

* What do you mean by the total? Please elaborate.

> - PDF export does not fit on the page

* We don't see any issue with the PDF export.  [Invoice_journal_report_Billings_MT.pdf](/uploads/974222035e6073966c6b0c43e29277b4/Invoice_journal_report_Billings_MT.pdf)

> - GUI layout causes issue in the header of the web page. Maybe put this in a window with a scroll bar at the bottom and top of the report? (Same this happens on the Revenue Analysis Worksheet report too)

* What's the issue? Can you please elaborate as we're not seeing any?

> - What is the reference column? it seems to be blank for everything (Same this happens on the Revenue Analysis Worksheet report too)

* This is a reference number of the account.  
  ![image](/uploads/42b96e29de1b16b648ff7e07d6804dea/image.png)

> - It does appear to be pulling the correct data in and the numbers seem to be accurate.

---------

**Detail Open AR:**  

> I thought this report was supposed to be able to select a date to run it for?

* We had already mentioned this on our last [comment] (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27488) that what we're doing & why?  
  * We did not added the date column till now becuase as mentioned on the report request by Gaylan that `Detail open AR report which is basically a listing of open invoices as of a month end or as of the date the report is run.` The report shows the details of the time when the report runs.

> Notice that it states "no longer includes pending and draft items". I thought it needed to include pending payments and draft credits?

* The question states same as ours [comment] (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27488).


> **ISSUE:**
>
> So I was going to attempt to make a payment to Account Demo 54 and then rerun the report to insure that it updates accordingly. But when trying to access the payment history page I received the something went wrong error: https://merge.sabilling.com/accounts/5b658fd2cf85c153b800000a/payments?sid=

* We have tried earlier to pay from the Verichek on this account due to which there was an error on the failed payments, albeit we have resolved the issue now.  

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 26, 2018, 03:40

added 1h of time spent at 2018-09-26

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 26, 2018, 06:33

@kdjstudios @pchaudhary Notes from our review:

**Invoice Journal Report**:
* PDF output is truncating the data. Pls try a landscape orientation and see if that fits.  
* The CSV output does have all columns.
* Tim will review the date range vs. billing cycle with Kyle.
* Tim will review the totals comment (I don't think we need totals).
* Re reference number: I'm OK removing this as long as it is not being being used by either Answernet or the hosted accounts. Can we please get a count of how many account do/do not use it?
* The fields 'Account', 'Reference' and 'Name' are repeated before the current invoice info - these can be removed.

**Detail Open AR:**
* This report should be run as of a date the user specifies.
* As of that date, we want to list the most recent unpaid invoice prior to the 'as of' date.  In other words, whatever the last invoice with an unpaid balance was prior to the 'as of' date.
* If there are any payments that were dated after the invoice, but before the 'as of' date of the report, then they need to be included.

@smasih Sumeet is going to recap what he thinks will get close in terms of calculating what was open at any point in time.  Tim will review & discuss.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 26, 2018, 08:41

@pchaudhary @smasih Hi Piyush and Sumeet, I did specify here http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_26481 that there is supposed to be a date range and that it should support multiple invoices with sub totals by account.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 26, 2018, 11:23

@smasih @pchaudhary 

Thank you for the updates and reviewing this with @tim.traylor too. I was also able to review with Tim and just want to post overall updates in one comment. Please let us know when this is ready to test again. We will want to have either by Monday or this Friday morning the reports ready to allow Keener to run and review them. 


**Invoice Journal Report**:

- We still don't get any confirmation about the **Invoice number & Date** on the Invoice Journal Report.

> - I thought this report was supposed to be run via a date range, not a billing cycle?

- If this is the case, then we have to re-estimate of this report as the function we're using is for the billing cycle as the details are already on the database for that billing cycle. **Please see http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_26481 and also the very first line in the original request states this.**

> - I see multiple lines for each account, but I don't see the totals for each account?

- What do you mean by the total? Please elaborate. **Please see http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_26481 as stated this can be for a date range and thus include multiple invoices. As Tim states we will then need to have subtotals for each account.** 

> - PDF export does not fit on the page

- We don't see any issue with the PDF export.  [Invoice_journal_report_Billings_MT.pdf](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/974222035e6073966c6b0c43e29277b4/Invoice_journal_report_Billings_MT.pdf) **Please see http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27609 and please try to use landscape in order to prevent the data from being truncated.**

> - GUI layout causes issue in the header of the web page. Maybe put this in a window with a scroll bar at the bottom and top of the report? (Same this happens on the Revenue Analysis Worksheet report too)

- What's the issue? Can you please elaborate as we're not seeing any? **Please See Screenshot Below. the header colors stop in the middle of the screen**
![image](/uploads/34c6fb371670c7dc939d7ffcbf7f18ac/image.png)

> - What is the reference column? it seems to be blank for everything (Same this happens on the Revenue Analysis Worksheet report too)

- This is a reference number of the account.  
    ![image](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/42b96e29de1b16b648ff7e07d6804dea/image.png)

> - It does appear to be pulling the correct data in and the numbers seem to be accurate.  **Thank you, Please confirm what percentage of accounts for keener actually have information in this field. If it is minimal, we will want to remove this field.**

-----

**Detail Open AR:**

> I thought this report was supposed to be able to select a date to run it for?

- Discussed with @tim.traylor in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27609

> Notice that it states "no longer includes pending and draft items". I thought it needed to include pending payments and draft credits?

- Discussed with @tim.traylor in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27609

> **ISSUE:**
>
> So I was going to attempt to make a payment to Account Demo 54 and then rerun the report to insure that it updates accordingly. But when trying to access the payment history page I received the something went wrong error: https://merge.sabilling.com/accounts/5b658fd2cf85c153b800000a/payments?sid=

- We have tried earlier to pay from the Verichek on this account due to which there was an error on the failed payments, albeit we have resolved the issue now. **Thank you, please let me know when this is available for testing**

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 27, 2018, 08:32

Hi [Kyle] ( @kdjstudios ),   

**Invoice Journal Report**:

> I thought this report was supposed to be run via a date range, not a billing cycle?

* Completed by the Date range.  


> A date range and thus include multiple invoices. As Tim states, we will then need to have subtotals for each account.

* Updated the total column as the total of that invoice of that row.


> Please try to use landscape in order to prevent the data from being truncated.

* We have removed many of the unneeded columns which reduce the width of the report. So, the PDF issue is automatically fixed.


> GUI layout causes an issue in the header of the web page. Maybe put this in a window with a scroll bar at the bottom and top of the report? (Same this happens on the Revenue Analysis Worksheet report too)

* We have removed many of the unneeded columns which reduce the width of the report. So, the layout issue is automatically fixed.  



> What is the reference column? it seems to be blank for everything (Same this happens on the Revenue Analysis Worksheet report too)

* **Internal** - 5214 out of 15402 accounts **does not** have the Reference number.  
  **External** - 3114 out of 3903 accounts **does not** have the Reference number.



**Detail Open AR**:  

* We still need an discussion for this report.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 27, 2018, 08:32

added 6h of time spent at 2018-09-27

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 27, 2018, 08:39

@pchaudhary @kdjstudios Thanks for the count Piyush.  Kyle, with the Answernet count it sounds like we should leave it in.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 27, 2018, 08:59

@pchaudhary @smasih @amishra @tim.traylor 

Thank you for the update. I just quickly ran the Invoice Journal Report on MERGE2 for SpeedEz and its looking great. I find the layout way more appealing now and the exports are looking better too. The date range functionality appears to be working wonderfully.

- I did find an issue on the PDF export that the headers on all pages except the first are not aligned correctly.

![image](/uploads/5462c49a68223cc120da78e0996ca021/image.png)

- We are having the an issue with the calendar display layout under the headers

![image](/uploads/353b67ed11656dc8e880940d58b2b297/image.png)

- Also there is what looks to be a popup on top of the calendar with a date that is blocking the user from seeing all options on the calendar too

![Untitled](/uploads/e953770202ae0f1fdcb092a7fc6c8981/Untitled.png)

- Very lastly, might just be a preference of mine, but can we remove the underscores from the end of the file names that are downloaded. They are named "invoice_journal__.pdf" or "invoice_journal__.csv" can we just have it be "invoice_journal.pdf" or "invoice_journal.csv"

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 28, 2018, 04:47

Hi [Kyle] ( @kdjstudios ),  

Thanks for your words.  

**Invoice Journal Report** :

> I did find an issue on the PDF export that the headers on all pages except the first are not aligned correctly.

* I looked into this & found that this is an ongoing issue with all of the reports, so we have to look into it more deeply. For that concern, we're creating another ticket to serve this purpose only which helps us to lead on this issue.  

> We are having an issue with the calendar display layout under the headers

* fixed

> Also there is what looks to be a popup on top of the calendar with a date that is blocking the user from seeing all options on the calendar too

* we are not able to replicate the same however, it doesn't seem a big issue. You can close that popup on clicking it.

> Very lastly, might just be a preference of mine, but can we remove the underscores from the end of the file names that are downloaded. They are named "invoice_journal__.pdf" or "invoice_journal__.csv" can we just have it be "invoice_journal.pdf" or "invoice_journal.csv"

* fixed.

**Detail Open AR** :

* We discussed it internally & found that this can only be possible by creating a daily snapshot of the accounts & its invoice & payments.  

* We have to create 2 models as below:
  * **Payment Invoice Mapping** - 
    * payment_id
    * invoice_id
    * amount_paid
  * **Account Detail AR** - 
    * account_id
    * account_balance
    * account_status
    * invoice_id
    * invoice_balance

From the above 2 models, we can calculate the all values needed of the account for the specific date.

We are going to discuss this with Tim on our today's huddle.

Thanks.


Internal Discussion: 0.5Hr
Development: 1Hr
Testing: 0.5Hr

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 28, 2018, 04:47

added 2h of time spent at 2018-09-28

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Sep 28, 2018, 09:38

@smasih 
* Cron jobs & crash reporting
* Script to select open invoices & write data to the new collection
* Update of new report to run off of the new collection
* Testing of the daily job and the report (need multiple runs over at least a week-long period).

Tim estimates 8 hours (3 or 4 of that is for the daily testing over a period).

Sumeet, please check this estimate and let me know if you see any issues with this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2018, 11:36

I wrote:

> The support team has been able to publish the Invoice Journal report for your review. We currently have this on a testing server for you to look over and provide any feedback. To access the report please follow the steps below. If you have any questions or concerns please let us know. 
> 
> 1. Open the testing server of SAB: https://merge2.sabilling.com/reports/InvoiceJournal
> 2. You should be able to login with your same username and password.
> 3. Once logged in, you should be presented directly with the Invoice Journal Report.
> 4. Please select your site, date range, and then click the "Go" button.
> 5. The report should display for you in a moment.
> 
> Please let us know what your thoughts are on the report; if we are missing any columns, metrics, values, etc; or simply if you have any additional requests to improve the report. The support team is still working on the Detail Open AR report and we hope to have that report ready for you to test and provide some initial feedback on early next week.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 1, 2018, 14:58

@smasih @pchaudhary @amishra @tim.traylor 

Gaylan wrote:

> I went to the merge2.sabilling.com/reports/invoiceJournal so I could review the new report that is in test mode, but I got the error We’re sorry, but something went wrong.
> 
>  
> 
> Also, can we make sure that Bill has access to this as well with his log in since he is the one that is actually going to be “testing” it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 3, 2018, 08:12

@smasih @pchaudhary @amishra @tim.traylor 

Any updates? Looks like Gaylan was unable to test the new report. Also any progress on the Detailed Open AR?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 3, 2018, 08:14

@kdjstudios 

As per our discussion with @tim.traylor he said we need to get an approval from Garry on the new [estimate](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27844) for this report.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 3, 2018, 08:36

@smasih 

Thanks, I do not see in the ticket where we are needing Gary's Approval. I will reach out to @tim.traylor

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 3, 2018, 11:24

@smasih @pchaudhary @amishra 

After discussing with @tim.traylor we are only waiting on approval from Gary for the second report. 

Please continue to work on the first report (Invoice journal) and insure that Gaylan is able to actually use it on MERGE2 so we may start receiving feedback.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 4, 2018, 11:34

@smasih @pchaudhary @amishra @tim.traylor 

Bill wrote:

> Hi Kyle,
>
> I am working on integrating the SA Billing process into our accounting systems and I’ve been working with Gaylan to get reports from SA Billing to help with this.  It makes since to take Gaylan out of the loop at this point and work directly with you.
> 
> I have reviewed the Invoice Journal report and my comments are below.  Let me know if you have any questions.

[SA_Billing_-_Invoice_Journal_-_BL_Comments_-_2018-10-03.xlsx](/uploads/2f2b7b400c7f6bea1352720fedff966e/SA_Billing_-_Invoice_Journal_-_BL_Comments_-_2018-10-03.xlsx)


(NOTE: I had to take a screenshot in order to keep the formatting/bulletpoints/color/etc)
![image](/uploads/e0801a11fb82a0e3233f68d515523446/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Oct 9, 2018, 06:59

Hi [Kyle] ( @kdjstudios ),  

Please see below for the answers asked by Bill:

> Issues:  
  1. When I ran the KNR report I only got 12 records. (*From: 09/01/2018 To: 09/30/2018*)

* The 12 records are found when the `Include Draft` option has been selected as `NO`. We have tried by choose as `Yes` & got **521 records** for the same date range.  

> a. Appears that I only got the records where the Account number was duplicated.  

* The above reply describes this as well.  

> b. For KNRVA I got the same records you sent.  

* Nothing to be done here.  

> c. Should the Test account be included?  

* We are listing all of the accounts for the selected Site.

> 2. The report on the screen includes Invoice# and Invoice Date, the Download the .csv report does not include these. They need to be added to the .csv report.

*  Fixed.

> 3. See the comparison of report totals to totals from Revenue Analysis Worksheet Report.  
  a. For KNR the Invoice Journal does not include "Usage" revenue.  

* We are not able to follow here. Could you please explain briefly.  

> b. For KNRVA the Invoice Journal "Charges" revenue is $44,350.59 vs $321.43 in the Revenue Analysis Worksheet. My guess is that the Revenue Analysis Worksheet is probably correct.

* Here, I would like to mention that the Revenue Analysis Worksheet report lists the invoices by the selected billing cycle and the Invoice Journal report lists the invoices by the invoice date of selected date range which may cause the difference of number.  

> 4. The following accounts are in the KNR Revenue Analysis Worksheet but not in the Invoice Journal Report.  

* Sorry to say but we're not able to see the accounts on the KNR Revenue Analysis Worksheet report of *09/23/2018 Master* Billing Cycle. I think it would be really helpful if you can provide us with the specific billing cycle you were running.

> a. Shouldn't the Invoice Journal report for a calendar month include all of the invoice cycles in that month?

* The Invoice Journal report selects the invoices of the site whose invoice date comes under the range of the selected range. If you need to change us the functionality then please elaborate to what to do?  

> 4. The revenue by type columns does not add up to the total column when there is an amount in the Credit column. It appears that the Total subtracts the  negative credit amount thereby adding it.   

* Do you need to show up the totals as the revenue analysis worksheet in the last of the report.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Oct 9, 2018, 07:00

added 1h of time spent at 2018-10-09

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 9, 2018, 08:22

I wrote: (Took Screenshot due to Gitlab's formatting mismatch)

![image](/uploads/6873eb142819fb02f746f3b62ab4f6ee/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Oct 10, 2018, 06:27

@kyle Gary needs to approve the daily snapshot concept for the report. Lets put it on the agenda 10/11.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2018, 09:54

Mentioned in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27761 the Detail Open AR will need to be discussed with Gary before we make further progress.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 19, 2018, 13:54

Bill Wrote:

> I am getting back to this issue but the link I used to access the Invoice Journal report 
> https://merge2.sabilling.com/reports/InvoiceJournal
> Is not working.  Please provide the link to the report so that I can review the status.

@smasih @pchaudhary @amishra @tim.traylor 

This should not be happening. We need to insure that we keep this available to the client for testing and feedback. If we need to move this to a different dev server please do so and update us accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Oct 23, 2018, 24:10

@kdjstudios 
I am transferring keener reports to `dev.sabilling.com` for their review.
This server is hardly used so it will stay there.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 23, 2018, 09:15

I wrote:

> We do apologize the support team had started using the server below for other developments. If you could please use the new link below to access the report and test it. 
> 
> https://dev.sabilling.com/reports/InvoiceJournal

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Oct 23, 2018, 09:34

The creation of the daily snapshot is approved.  Please proceed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 30, 2018, 15:22

Bill wrote:

> My regular user name & password is not working at this site, is there a different account I should be using?

@smasih @amishra @pchaudhary @tim.traylor 

May we please check to confirm you copied the user login data over to this server?

EDIT: We may also want to insure that we have copied over updated site information and data to insure the client can look at current metrics.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Oct 31, 2018, 06:36

@kdjstudios Hi Kyle - the Open A/R is ready for review as well.  Can you pls pls take a look and if all is well pass it along tot he client for review?  Thx

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Oct 31, 2018, 08:20

Hi [Kyle] ( @kdjstudios ),  

There are other items to be tested as well so once I merged all of them into one branch then I'll let you know.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 31, 2018, 08:25

@pchaudhary 

Thank you for the update, please let me know once your testing has been completed. Please also make sure the client's users have access to these reports so we may get their testing and feedback as well.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Oct 31, 2018, 11:25

@kdjstudios 

We have completed the report development please test it, and get a feedback from Client.

Currently, we have data for 2 days (29 and 30) and its a test server so there is no change b/w data entries.

The system will run a snapshot process every day at 12am

other that we can get the same data if we select the live option.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 1, 2018, 14:33

Thank you. I checked the reports over and they appear to be functioning fine for me. I don't see any issues as of right now.

I wrote:

> The support team has updated the reports to be available at the following links below. In regards to the Detail Open AR, the report required for SA Billing to take 'snapshots' of account information at the end of each day; which SA Billing was not doing this before 10-29-18. The report will only work from that date and forward. Please let us know if you have any question or concerns. 
> 
> Detail Open AR:
> https://merge2.sabilling.com/reports/DetailOpenAR
> 
> Invoice Journal:
> https://merge2.sabilling.com/reports/InvoiceJournal

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 5, 2018, 08:16

Bill wrote:


> RE Detail Open AR:
> 
> 1. Overall the report looks good, it has the right columns.
> 2. For KNR the only “Snapshot” date available was 11/2/2018.  The total of this report was $772,884.64.  The standard “AR Snapshot” report as of 10/31 totaled $342,884.08 and adjusted for payments posted on 11/1 and 11/2 the total as of 11/2 should be $338,744.11.  The balance for the Open AR report is more than twice what it should be.  Part is due to many of the invoice being duplicated and part due to many of the Client Account numbers in the AR Snapshot report not being in the Open AR report.
> 3. Similarly the “Live” report as of 11/3 balance was $386,442.32 which is ½ of the 11/2 Snapshot version.
> 4. The CSV download function did not work.
> 5. Key for this report is that it should have the same balance as the standard “AR Snapshot” report as of month end.
>  
> 
> RE Invoice Journal:
> 
> 1. Running the report for the date range 10/1 to 10/31/18 for KNR only resulted in 37 records.  Note that Include Draft = “No” which is appropriate.  There should be about 530 invoices in October.
>  - It appears that the report only includes invoices that have amounts in “Charges” and “Credits” fields.  The other billing categories (Recurring, Usage, Periodic & LateChgs) are all $0.
> 3. In the report the amounts in the Credits column are shown as negative however the Total column includes these amounts as positive numbers.  I suspect that the formula for the Total column subtracts the Credits field when it should just add it as it is already negative.
> 4. All of the invoice dates in the Invoice Journal report are 10/22/18 which is not the same as the invoice dates per the standard “Invoices By Site” report for the 10/21/2018 Billing Cycle.  e.g. invoice #KNR-30923 is dated 10/12/18.
> 5. Key for this report is that should total the same as the standard Revenue Analysis report for the Current Month and include the detail of the invoices issued in the month, not just the total by Account.

I wrote:

> Thank you for your feedback. I should have my replied to your statements below. In regards to the items below that will need updates from our support team we will post updates to them accordingly. 
> 
> 
> RE Detail Open AR:
> 
> Overall the report looks good, it has the right columns. 
> - This is great to hear. Thank you.
> 
> For KNR the only “Snapshot” date available was 11/2/2018.  The total of this report was $772,884.64.  The standard “AR Snapshot” report as of 10/31 totaled $342,884.08 and adjusted for payments posted on 11/1 and 11/2 the total as of 11/2 should be $338,744.11.  The balance for the Open AR report is more than twice what it should be.  Part is due to many of the invoice being duplicated and part due to many of the Client Account numbers in the AR Snapshot report not being in the Open AR report.
> - These reports are on a development server. Our development servers do not generally have the exact same 'accounting/billing' info that the production server does. I believe we stated this when the report was first release, but I am unsure if you were part of that email yet. I will reach out to the support team to see if they are able to copy over the production servers data so you can compare and contract the metrics accordingly. 
> 
> Similarly the “Live” report as of 11/3 balance was $386,442.32 which is ½ of the 11/2 Snapshot version.
> - Same response as above. The development servers do not generally contain the same information as your production server. 
> 
> The CSV download function did not work.
> - Thank you for this feedback. The support team will be looking into this. We will follow up once this is functional.
> 
> Key for this report is that it should have the same balance as the standard “AR Snapshot” report as of month end.
> - Agreed. I have informed the support team, and do believe we will need to have them either release this to the production server so you may confirm the results match, or have them copy the production servers data over to the development server for comparison. We will follow up accordingly. 
> 
> RE Invoice Journal:
> 
> Running the report for the date range 10/1 to 10/31/18 for KNR only resulted in 37 records.  Note that Include Draft = “No” which is appropriate.  There should be about 530 invoices in October.
> - In my email sent to you; we stated this report will only have data from the 10-29 and forward. This is due to the report requiring a daily snapshot, which SA Billing was not taking before this report was developed. You will not have any results before the 29th of October. 
> 
> It appears that the report only includes invoices that have amounts in “Charges” and “Credits” fields.  The other billing categories (Recurring, Usage, Periodic & LateChgs) are all $0.
> - I would believe this is related to the above response as well. 
> 
> In the report the amounts in the Credits column are shown as negative however the Total column includes these amounts as positive numbers.  I suspect that the formula for the Total column subtracts the Credits field when it should just add it as it is already negative.
> - Thank you, we have informed the support team and will provide you with an update accordingly. 
> 
> All of the invoice dates in the Invoice Journal report are 10/22/18 which is not the same as the invoice dates per the standard “Invoices By Site” report for the 10/21/2018 Billing Cycle.  e.g. invoice #KNR-30923 is dated 10/12/18.
> - As stated in previous emails and responses above. The data on the Development server is not the same as the Production server. We will discuss with the support team to either copy the data from production over or to release these onto production for your review there. 
> 
> Key for this report is that should total the same as the standard Revenue Analysis report for the Current Month and include the detail of the invoices issued in the month, not just the total by Account.
> - As stated in previous emails and responses above. The data on the Development server is not the same as the Production server. We will discuss with the support team to either copy the data from production over or to release these onto production for your review there. 

@smasih @amishra @pchaudhary @tim.traylor 

Judging from the intial response, I would believe we need to confirm two things and then release this onto production so the validity of the metrics can be confirmed.

1. Fix the CSV download.
2. Release onto production, with access only to us Tech, and those whom have access to Keener sites. 
 - This way we can confirm the metric values and also have Keener confirm too.
 - I believe this would be quicker, and more accurate way then copying data to the development servers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 13, 2018, 09:47

This was discussed today. we will need the following done:

1. Fix the CSV.
2. Release onto production with a "BETA" Label.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Nov 14, 2018, 05:54

@kdjstudios Kyle, the team will do a special release for this and it will be available to the client tomorrow AM.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2018, 12:33

I wrote:

> After discussion with the support team they will have both reports available on your production server that way you will be able to test and confirm the metrics with live data accordingly. We plan to have this released for you tomorrow morning.

EDIT: Please be sure to update the ticket labels, once this has been released. Thanks!

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 15, 2018, 01:29

Hi [Kyle] ( @kdjstudios ),  

* We have successfully released both of the reports on both of the production servers.( *Internal & Hosted*)  
* We have also added the *BETA* label on the report page.  
* The Detail Open AR report will run daily at 11:59 P.M.  
* We ran it for today manually & it'll automatically continue from tomorrow onwards.  

We'll wait for the review from Gaylan.  

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 15, 2018, 08:31

@pchaudhary @amishra @smasih @tim.traylor 

Thank you, 

First, The Invoice Journal report does not have a BETA label anywhere on it or on its link in the drop down.

Secondly, I see the Open AR report has the Beta Label once your on the report page itself. However, I think that there may have been lack of clarification but I believe @tim.traylor had mentioned to have this be on the link in the drop down too. Please bring this up for clafication with @tim.traylor. In addtition, why were we not running the daily capture since the 29th like we did on the other server? This report is now almost worthless for at least a few days to a month before the client can utilize it; since there is no historical data.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 16, 2018, 05:18

@kdjstudios 


> First, The Invoice Journal report does not have a BETA label anywhere on it or on its link in the drop down.

* Sorry for that. We missed this at the time of release, however, it's fixed now.


> Secondly, I see the Open AR report has the Beta Label once you are on the report page itself. However, I think that there may have been lack of clarification but I believe @tim.traylor had mentioned having this be on the link in the drop down too. Please bring this up for clarification with @tim.traylor.  

* We also earlier thought about that and also tried to put it there but then we found that for that only we've to spend more time because the option we choose on the drop down is directly considered as the URL, so for that, we need to change the report name also in the backend which needs so many other manipulations in code that's the reason we dropped the idea.



> In addition, why were we not running the daily capture since the 29th as we did on the other server? This report is now almost worthless for at least a few days to a month before the client can utilize it; since there is no historical data.  

* Because, the model cron job is using was not released on the production server at that time. So, there was no use to put it that there.  

Hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 16, 2018, 11:47

I wrote:

> We are just following up to let you know the reports are available on your production server for your review. They currently are labeled 'BETA' until you have approved them as being complete. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 23, 2018, 09:37

@smasih @amishra @pchaudhary @tim.traylor 

Please review these concerns and have them corrected. Please let us know if you have any questions. 

Bill wrote:

> I have accessed the reports and I have the following comments:
> 
> Detail Open AR
> 
> * The report appears to work and download ok EXCEPT that each invoice is included twice.
> * Will need to test at month end to confirm that the balance is the same as the AR Snapshot report
> 
> 
> Invoice Journal
> 
> * The report screen for this report is titled “Detail Open AR BETA”, it should be “Invoice Journal”
> * The format of the report looks good and it downloaded ok.
> * There are calculation errors in the report:
> * The Charges column totaled for 10/21 cycle (10/22 invoice date) $251,169.40.  Per the Revenue Analysis Worksheet the balance should be $5,321.86.
> * The Total column is still subtracting the Credits column when it should be adding it. 
>  
> 
> Please let me know when these issues have been fixed and I will re-test.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 26, 2018, 01:26

Hi [Kyle] (@kdjstudios ),   

We're going to create 2 new tickets (#1254 & #1255 ) for each of the report separately.  

As the ticket is serving for long 2 months & now the comment section becomes so long which can create lots of confusion by going on each comment and extract the part of a particular report.

We are also considering Tim's word, as he once suggest us to create a new ticket if any feedback comes after the release on production.

It'll also help us to serve each ticket separately.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 26, 2018, 09:20

Moving back to development as we have to open two new tickets to resolve this one client request. Please be sure to put all hours in the main ticket so we know how long this truly did take to create.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2018, 08:46

@smasih @pchaudhary @tim.traylor @amishra 

Bill wrote:

> I have run the new reports as of Nov month end and other than changing the Invoice Journal report screen to reference the right report all of the issues I identified below continue.  Let me know when you expect to have these addressed.  I am available for a call to clarify any of my comments.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 3, 2018, 13:25

I wrote:

> Thank you for reaching out for an update. The support team is currently testing the fixes and we hope to have these ready for you testing soon. I currently do not have an ETA from the support team, but will keep you updated accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 6, 2018, 09:49

@pchaudhary @smasih @amishra @tim.traylor 

We need to have these go live by Saturday. This mean with final approval from the client they are working; not just that they are on production server. Please insure they are working correctly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 17, 2018, 16:24

I wrote:

> We are just following up to confirm if there was any additional feedback on the reports. If not, we will take the 'beta' label off of them. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 24, 2018, 09:13

Bill Wrote:

> I got a chance to try the new reports and they look great.  The tie to other invoice related reports and the issues I had all appear to be resolved.
>
> Thanks for your great work.

@smasih @pchaudhary @amishra 

Please remove the 'Beta' label from both reports. Then we can mark this ticket as closed.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 27, 2018, 01:18

@kdjstudios 

We have removed the label from both reports.   


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2018, 09:18

@amishra 

Thank you, closing the ticket.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2018, 09:18

closed
